packages/react-core/src/components/Drawer/DrawerPanelContent.tsx

```diff
@@ -261,10 +261,12 @@ export const DrawerPanelContent: React.FunctionComponent<DrawerPanelContentProps
           )}
           ref={panel}
           onTransitionEnd={(ev) => {
-            if (!hidden && ev.nativeEvent.propertyName === 'transform') {
-              onExpand();
+            if ((ev.target as HTMLElement).id === (id || panelId)) {
```

Comment:
An alternative to this:

```suggestion
            if ((ev.target as HTMLElement) === panel.current) {
```

Not really sure how common an issue it'd be, but one thing I noticed when checking if the target id === the component id/panelId is that if we have something like the following:

```
<DrawerPanelContent id="tester">
```

Then if within the drawer if some element like an expandable toggle would be the target *and* have that same ID, it causes the original bug to resurface. In the clip below I just added `id="tester"` to `.pf-c-expandable-section__toggle-icon` via dev tools since I don't believe we could pass an ID to that exact element otherwise, but I'm more imagining a scenario where some custom expandable toggle is used instead.


https://user-images.githubusercontent.com/70952936/225942700-22a5cbbb-e6eb-4223-9c5f-b61bb9b50e06.mov

Checking whether the target === the panel.current with the same setup as the above clip:


https://user-images.githubusercontent.com/70952936/225943008-1151c3a7-57b5-4243-9501-de06a4a0bcbc.mov

Personally I might lean more towards checking the target against the panel ref since ID duplication can happen, but like I said I'm not sure how common this would actually be. What do you think, @MariaAga ?

Reply:
Yeah I think its better to be safer, changed it